rustls/src/lib.rs

```diff
@@ -424,6 +465,16 @@
 #![cfg_attr(bench, feature(test))]
 #![no_std]
 
+// This constraint is also enforced by `portable-atomic` crate - enforcing here as well for extra clarity
+// (with a QUICK WORKAROUND)
+#[cfg(all(
+    feature = "critical-section",
+    feature = "unsafe-assume-single-core",
+    // QUICK WORKAROUND for issue with CI semver job:
+    not(feature = "std"),
```

Comment:
I think a quick note should be added concerning the potential effects of this workaround.

As I said elsewhere I think this is something not to be too concerned about as the likelihood of `unsafe-assume-single-core` working together with `std` would be a really weird fluke, anyways. I think a note with just this info should be good enough.